app/services/search_service.rb

```diff
@@ -88,8 +87,8 @@ def perform
     @result = if q.present?
                 ExactSearch.new(q, as_of).search!.presence ||
                   FuzzySearch.new(q, as_of).search!.presence
-              end
+              end || NullSearch.new(q, as_of)
 
-    @result ||= NullSearch.new(q, as_of)
+    @result
   end
```

Comment:
rubocop requirement, it suggested to use 
"@perform= ..." which I do not like, 
I've done this change to avoid that, otherwise a bigger refactor would be required (which is a good idea, but not now)